spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -80,6 +80,8 @@
 
 	private volatile Runnable stopCallback;
 
+	private volatile boolean readFromLeft = false;
```

Comment:
I guess it depends how much we want the abstraction to leak, but given that we are defining an attribute on the *redis* adapter, it's probably more intuitive if we use the Redis terminology of right/left. Likewise if we go with Gary's suggestion (I do like the succinctness of the single word attributes), we probably should use "pop" instead of "pull" on the inbound side.